src/governance/governance-classes.cpp

```diff
@@ -690,8 +687,8 @@ bool CSuperblock::IsExpired()
         CGovernanceObject* pgovobj = GetGovernanceObject();
         if (pgovobj) {
             LogPrint(BCLog::GOBJECT, "CSuperblock::IsExpired -- Expiring outdated object: %s\n", pgovobj->GetHash().ToString());
-            pgovobj->fExpired = true;
-            pgovobj->nDeletionTime = GetAdjustedTime();
+            pgovobj->SetExpired();
+            pgovobj->PrepareDeletion(GetAdjustedTime());
```

Comment:
This previously didn't check `nDeletionTime == 0` but now it does in PrepareDeletion. Not sure if this matters here or if there's any way it could not be 0.

Reply:
Hmmm... indeed. And it probably should check it because otherwise bumping `nDeletionTime` could extend the life of the object every time we call `IsExpired()` which sounds like a bug actually.

Reply:
Thinking more about it, this whole block of code which _updates_ the state of the object while calling `IsExpired()` is weird... hmm..

Reply:
Dropped 38b1ad7d0 completely. Will submit another PR which reworks this.